addons/website_sale/models/website_snippet_filter.py

```diff
@@ -196,3 +196,10 @@ def _get_products_alternative_products(self, website, limit, domain, context):
                     display_default_code=False,
                 ).search(domain, limit=limit)
         return products
+
+    def _prepare_values(self, limit=None, search_domain=None):
+        website = self.env['website'].get_current_website()
+        if website.ecommerce_access == 'logged' and self.env.user._is_public() and self.model_name == 'product.product':
```

Comment:
what if a model is `product.template' ?

Reply:
If I see corectly even when variants are disabled, both product template and product product is created and I suppose either way snippet takes product_product as a value.